### PR TITLE
xplat: fixes/improvements to PAL init and tracing

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

File: pal/src/memory/heap.cpp [hunk 1]
```diff
@@ -150,6 +150,13 @@ HeapCreate(
 }
 
 
+__attribute__((no_instrument_function, noinline))
+static bool PAL_Initialize_Check_Once()
+{
+    int error = PAL_InitializeChakraCore();
+    return error == ERROR_SUCCESS;
+}
```
Comment (jianchun) [inline pal/src/memory/heap.cpp]: Why is this code repeated in different files?

**Comment by jianchun** [comment]
:shipit:

File: pal/src/memory/heap.cpp (hunk 1 above)
Comment (obastemur) [inline pal/src/memory/heap.cpp]: I didn't want this function is available externally or dependent to another binary or exec. order.

**Comment by obastemur** [comment]
@jianchun Thanks for the review

**Pull request merged**